File: src/main/java/com/flanks255/simplylight/blocks/LampBase.java
package com.flanks255.simplylight.blocks;

import com.google.common.base.Suppliers;
import net.minecraft.resources.ResourceLocation;
import net.minecraft.world.level.block.Block;
import net.minecraftforge.registries.ForgeRegistries;

import java.util.function.BiConsumer;
import java.util.function.Supplier;

public class LampBase extends Block {
    public LampBase(Properties props) {
        super(props);
    }

    private final Supplier<ResourceLocation> lazyRes = Suppliers.memoize(() -> ForgeRegistries.BLOCKS.getKey(this));

    public ResourceLocation getRegistryName() {
        return lazyRes.get();
    }


    public void addLang(BiConsumer<String, String> consumer) {
    }
}

File: src/main/kotlin/com/flanks255/simplylight/blocks/LampBase.kt
package com.flanks255.simplylight.blocks

import com.flanks255.simplylight.simplylight
import net.minecraft.block.Block
import net.minecraft.block.material.Material
import net.minecraft.block.state.IBlockState
import net.minecraft.creativetab.CreativeTabs
import net.minecraft.item.Item
import net.minecraft.item.ItemBlock
import net.minecraft.util.math.BlockPos
import net.minecraft.world.IBlockAccess

@Suppress("OverridingDeprecatedMember")

open class LampBase(val name: String): Block(Material.REDSTONE_LIGHT){
    init {
        super.setLightLevel(1.0f)
        super.setLightOpacity(0)
        super.setHardness(0.5f)
        super.setUnlocalizedName(simplylight.MODID+"."+name)
        super.setHarvestLevel("pickaxe", 0)

        setRegistryName(name)
    }

    fun createItemBlock(): Item {
        return ItemBlock(this).setRegistryName(registryName)
    }

    override fun setCreativeTab(tab: CreativeTabs): LampBase {
        super.setCreativeTab(tab)
        return this
    }

    override fun getLightValue(state: IBlockState?, world: IBlockAccess?, pos: BlockPos?): Int = 15
}